example/randomCells2D/src/randomCells2D.cpp

```diff
@@ -0,0 +1,187 @@
+    using BufAccRandVec = alpaka::Buf<Acc, alpaka::rand::Philox4x32x10Vector<Acc>, Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+
+    constexpr Idx numX = NUM_X;
+    constexpr Idx numY = NUM_Y;
+
+    const Vec extent(numY, numX);
+
+    constexpr Idx perThreadX = 1;
+    constexpr Idx perThreadY = 1;
+
+    WorkDiv workdiv{alpaka::getValidWorkDiv<Acc>(
+        devAcc,
+        extent,
+        Vec(perThreadY, perThreadX),
+        false,
+        alpaka::GridBlockExtentSubDivRestrictions::Unrestricted)};
+
+    // Setup buffer.
+    BufHost bufHostA{alpaka::allocBuf<float, Idx>(devHost, extent)};
+    float* const ptrBufHostA{alpaka::getPtrNative(bufHostA)};
+    BufAcc bufAccA{alpaka::allocBuf<float, Idx>(devAcc, extent)};
+    float* const ptrBufAccA{alpaka::getPtrNative(bufAccA)};
+
+    BufHost bufHostB{alpaka::allocBuf<float, Idx>(devHost, extent)};
+    float* const ptrBufHostB{alpaka::getPtrNative(bufHostB)};
+    BufAcc bufAccB{alpaka::allocBuf<float, Idx>(devAcc, extent)};
+    float* const ptrBufAccB{alpaka::getPtrNative(bufAccB)};
+
+    BufHostRand bufHostRandA{alpaka::allocBuf<alpaka::rand::Philox4x32x10<Acc>, Idx>(devHost, extent)};
+    BufAccRand bufAccRandA{alpaka::allocBuf<alpaka::rand::Philox4x32x10<Acc>, Idx>(devAcc, extent)};
+    alpaka::rand::Philox4x32x10<Acc>* const ptrBufAccRandA{alpaka::getPtrNative(bufAccRandA)};
+
+    BufHostRandVec bufHostRandB{alpaka::allocBuf<alpaka::rand::Philox4x32x10Vector<Acc>, Idx>(devHost, extent)};
+    BufAccRandVec bufAccRandB{alpaka::allocBuf<alpaka::rand::Philox4x32x10Vector<Acc>, Idx>(devAcc, extent)};
+    alpaka::rand::Philox4x32x10Vector<Acc>* const ptrBufAccRandB{alpaka::getPtrNative(bufAccRandB)};
+
+    InitRandomKernel initRandomKernel;
+    alpaka::exec<Acc>(queue, workdiv, initRandomKernel, extent, ptrBufAccRandA);
+    alpaka::wait(queue);
+
+    alpaka::exec<Acc>(queue, workdiv, initRandomKernel, extent, ptrBufAccRandB);
+    alpaka::wait(queue);
+
+    for(Idx x = 0; x < numX; ++x)
+    {
+        for(Idx y = 0; y < numY; ++y)
+        {
+            ptrBufHostA[y * numX + x] = 0;
+            ptrBufHostB[y * numX + x] = 0;
+        }
+    }
+
+    alpaka::memcpy(queue, bufAccA, bufHostA, extent);
+    RunTimestepKernelSingle runTimestepKernelSingle;
+    alpaka::exec<Acc>(queue, workdiv, runTimestepKernelSingle, extent, ptrBufAccRandA, ptrBufAccA);
+    alpaka::memcpy(queue, bufHostA, bufAccA, extent);
+
+
+    alpaka::memcpy(queue, bufAccB, bufHostB, extent);
+    RunTimestepKernelVector runTimestepKernelVector;
+    alpaka::exec<Acc>(queue, workdiv, runTimestepKernelVector, extent, ptrBufAccRandB, ptrBufAccB);
+    alpaka::memcpy(queue, bufHostB, bufAccB, extent);
+    alpaka::wait(queue);
+
+    float avgA = 0;
+    float avgB = 0;
+    for(Idx x = 0; x < numX; ++x)
+    {
+        for(Idx y = 0; y < numY; ++y)
+        {
+            avgA += ptrBufHostA[y * numX + x];
+            avgB += ptrBufHostB[y * numX + x];
+        }
+    }
```

Comment:
Isn't this just an unnecessary 2D version of `std::reduce(ptrBufHostA, ptrBufHostA + numX * numY)`?

Reply:
Yes, it is. I'll use std::reduce in the 1D version. The 2D version is now more of a placeholder for some more interesting algorithm that would actually show something happening in 2D. I haven't figured out a nice simple but interesting example yet.

Reply:
In the end, I won't use `std::reduce` even in the 1D version yet. The *hemera* toolchain seems convinced that `error: namespace "std" has no member "reduce"` 😢

Reply:
Funny. I had no problems with `std::reduce` when I ran on hemera 3 weeks ago with gcc 10.2 and C++17.

Reply:
thx. I was on gcc/9 because of cuda. I've somehow missed that I can go cuda/11.2 + gcc/10.2